docs/pallas/tpu/core_map.md

```diff
@@ -0,0 +1,358 @@
+# Pallas Core-specifc Programming
+
+In this guide, we explore using `pl.core_map` to write Pallas kernels. Compared with `pallas_call`, `core_map` offers a few key characteristics:
+
+* **Per-core level programming**: You write code for an TPU/GPU core, not for a JAX device. This is crucial if you want to specifically control a core, or how cores communicate and distribute work among one another.
+
+* **Flexible pipelining**: You have the option to write pipelining communications on your own, instead of relying on Pallas grids and specs. This is helpful if your pipeline diverges from the standard "copy-in, compute & copy-out" pattern.
+
+* **Collectives**: Since `core_map` allows inter-core communications, it is especially helpful when writing collectives on the core level.
+
+This guide focuses on TPU. For how to use `core_map` on GPU to achieve higher thread flexibility, check out our [Pallas GPU `core_map` tutorial](https://docs.jax.dev/en/latest/pallas/gpu/reference.html#using-core-map).
+
+## Environment setup
+
+Modern accelerators often have multiple cores under a device. For TPU chips higher than v4, every JAX device by default contains two TensorCores (aka. a [Megacore](https://cloud.google.com/tpu/docs/system-architecture-tpu-vm#chips)). They also contain a [SparseCore](https://cloud.google.com/tpu/docs/system-architecture-tpu-vm#sparsecore), consisting of many subcores.
```

Comment:
Only v4 and v5p have megacore. v7 lacks megacore and instead comes in pairs of chips with separate HBM. Also I think only v5p/v6/v7 have sparsecore.

Reply:
Thanks for clarifying!